test_conformance/api/test_queries.cpp

```diff
@@ -476,30 +478,180 @@ REGISTER_TEST(get_command_queue_info_compatibility)
 REGISTER_TEST(get_context_info)
 {
     int error;
+
+    // query single device context and perform object comparability test
+    cl_uint num_devices = 0;
+    error = clGetContextInfo(context, CL_CONTEXT_NUM_DEVICES, sizeof(cl_uint),
+                             &num_devices, nullptr);
+    test_error(error, "clGetContextInfo CL_CONTEXT_NUM_DEVICES failed\n");
+
+    test_assert_error(num_devices == 1,
+                      "Context must contain exactly one device\n");
+
+    cl_device_id comp_device = nullptr;
+    error = clGetContextInfo(context, CL_CONTEXT_DEVICES, sizeof(cl_device_id),
+                             &comp_device, nullptr);
+    test_error(error, "clGetContextInfo CL_CONTEXT_DEVICES failed\n");
+
+    test_assert_error(device == comp_device,
+                      "Unexpected result returned by CL_CONTEXT_DEVICES query");
+
+    // query context properties against valid size
     size_t size;
     cl_context_properties props;
+    error = clGetContextInfo(context, CL_CONTEXT_PROPERTIES, sizeof(props),
+                             &props, &size);
+    test_error(error, "Unable to get context props");
 
-    error = clGetContextInfo( context, CL_CONTEXT_PROPERTIES, sizeof( props ), &props, &size );
-    test_error( error, "Unable to get context props" );
-
-    if (size == 0) {
+    if (size == 0)
+    {
         // Valid size
         return 0;
-    } else if (size == sizeof(cl_context_properties)) {
+    }
+    else if (size == sizeof(cl_context_properties))
+    {
         // Data must be NULL
-        if (props != 0) {
+        if (props != 0)
+        {
             log_error("ERROR: Returned properties is no NULL.\n");
             return -1;
         }
         // Valid data and size
         return 0;
     }
     // Size was not 0 or 1
-    log_error( "ERROR: Returned size of context props is not valid! (expected 0 or %d, got %d)\n",
-              (int)sizeof(cl_context_properties), (int)size );
+    log_error("ERROR: Returned size of context props is not valid! (expected 0 "
+              "or %d, got %d)\n",
+              (int)sizeof(cl_context_properties), (int)size);
     return -1;
 }
 
+REGISTER_TEST(get_context_info_mult_devices)
+{
+    cl_int err = CL_SUCCESS;
+    size_t size = 0;
+
+    // query multi-device context and perform objects comparability test
+    err = clGetDeviceInfo(device, CL_DEVICE_PARTITION_PROPERTIES, 0, nullptr,
+                          &size);
+    test_error_ret(err, "clGetDeviceInfo failed", TEST_FAIL);
```

Comment:
Minor: this could be shortened slightly by using `test_error_fail` rather than `test_error_ret` and explicitly returning `TEST_FAIL`:

```suggestion
    test_error_fail(err, "clGetDeviceInfo failed");
```

Same pattern is used in a few other places in this test.

Reply:
Corrected